nucliadb/nucliadb/ingest/orm/entities.py

```diff
@@ -95,16 +104,28 @@ async def get_entities_groups(self) -> dict[str, EntitiesGroup]:
 
     async def list_entities_groups(self) -> dict[str, EntitiesGroupSummary]:
         groups = {}
-        async for group in self.iterate_entities_groups_names(exclude_deleted=True):
-            stored = await self.get_stored_entities_group(group)
-            if stored is not None:
-                groups[group] = EntitiesGroupSummary(
-                    title=stored.title, color=stored.color, custom=stored.custom
+        max_simultaneous = asyncio.Semaphore(10)
+
+        async def _composition(group: str):
+            async with max_simultaneous:
+                stored = await self.get_stored_entities_group(group)
```

Comment:
Is this parallelization really needed? The method only gets a value from maindb

Reply:
it's doing a get for every group! we have 18 groups that learning detects, plus if the user enters their own, this could be a lot of gets in the end. 

IMO it doesn't harm